src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1481,30 +1481,20 @@ void uct_dc_mlx5_ep_pending_purge(uct_ep_h tl_ep,
 {
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_dc_mlx5_iface_t);
     uct_dc_mlx5_ep_t *ep       = ucs_derived_of(tl_ep, uct_dc_mlx5_ep_t);
+    uint8_t dci                = ep->dci;
     void *priv_args[2]         = {ep, arg};
     uct_purge_cb_args_t args   = {cb, priv_args};
     ucs_arbiter_t *waitq;
     ucs_arbiter_group_t *group;
     uint8_t pool_index;
-    uint8_t dci;
-
-    if (uct_dc_mlx5_iface_is_dci_rand(iface)) {
-        ucs_arbiter_group_purge(uct_dc_mlx5_iface_tx_waitq(iface),
-                                uct_dc_mlx5_ep_rand_arb_group(iface, ep),
-                                uct_dc_mlx5_ep_arbiter_purge_cb, &args);
-        return;
-    }
 
     uct_dc_mlx5_get_arbiter_params(iface, ep, &waitq, &group, &pool_index);
     ucs_arbiter_group_purge(waitq, group, uct_dc_mlx5_ep_arbiter_purge_cb,
                             &args);
 
-    if (ep->dci == UCT_DC_MLX5_EP_NO_DCI) {
-        return;
-    }
-
-    dci = ep->dci;
-    if (!uct_dc_mlx5_iface_dci_detach(iface, ep)) {
+    if ((dci == UCT_DC_MLX5_EP_NO_DCI) ||
```

Comment:
maybe
if (dci != NO_DCI && !is_rand && detach) { schedule_release }
and similar refactor (+remove dci helper var) in uct_dc_mlx5_iface_dci_do_dcs_pending_tx
BTW maybe we can move schedule_release inside detach ?

Reply:
good idea, done